pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1491,9 +1460,21 @@ func (nc *Controller) evictPods(node *v1.Node, pods []*v1.Pod) (bool, error) {
 	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID)), nil
 }
 
-func (nc *Controller) markNodeForTainting(node *v1.Node) bool {
+func (nc *Controller) markNodeForTainting(node *v1.Node, status v1.ConditionStatus) bool {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
+	if status == v1.ConditionFalse {
```

Comment:
Maybe we don't need judgment of status there.
The use of  `markNodeForTainting` will judge the status(in func `processTaintBaseEviction`).

Reply:
This judgement is imported from upstream kubernetes. you can reference from pull request here: https://github.com/kubernetes/kubernetes/pull/89059